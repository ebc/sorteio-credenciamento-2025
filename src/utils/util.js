// Retorna inteiro randômico no intervalo [min, max)
const getRandomInt = (min, max) => {
  return Math.floor(Math.random() * (max - min)) + min;
};

const sorteados2CSV = sorteados => {
  let ret = 'Lote,Inscrição,CNPJ,Razão Social\n';
  ret += sorteados.reduce(
    (consolidado, e, indice) =>
      (consolidado +=
        indice +
        1 +
        ',' +
        e['Inscricao'] +
        ',' +
        e['CNPJ'] +
        ',' +
        e['RazaoSocial'] +
        '\n'),
    ''
  );
  return ret;
};

const baixeResultados = (sorteados, log, dataSorteio) => {
  const blobSorteados = new Blob([sorteados2CSV(sorteados)], {
    type: 'text/csv;charset=utf-8'
  });
  const blobLog = new Blob([log], { type: 'text/plain;charset=utf-8' });

  var linkSorteados = document.createElement('a');
  var linkLog = document.createElement('a');
  const dataSorteioNormalizada = dataSorteio.replace(' ', '').replace('/', '-');
  linkSorteados.download = 'resultado_' + dataSorteioNormalizada + '.csv';
  linkLog.download = 'log_' + dataSorteioNormalizada + '.txt';

  var urlSorteados = (linkSorteados.href = URL.createObjectURL(blobSorteados));
  var urlLog = (linkLog.href = URL.createObjectURL(blobLog));

  linkSorteados.click();
  URL.revokeObjectURL(urlSorteados);

  linkLog.click();
  URL.revokeObjectURL(urlLog);
};

module.exports = {
  getRandomInt,
  sorteados2CSV,
  baixeResultados
};
