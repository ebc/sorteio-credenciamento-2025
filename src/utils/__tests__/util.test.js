import { getRandomInt, sorteados2CSV, baixeResultados } from '../util';

it('gera um inteiro randÃ´mico no intervalo [min, max)', () => {
  const min = 100;
  const max = 200;
  const inteiro = getRandomInt(100, 200);
  expect(inteiro).toBeGreaterThanOrEqual(min);
  expect(inteiro).toBeLessThan(max);
});

it('gerar string CSV a partir da lista de sorteados', () => {
  const sorteados = [
    { 'Inscricao': '123', 'CNPJ': '48797752000151', 'RazaoSocial': 'Produtora xyz' }
  ];
  const csv = sorteados2CSV(sorteados);
  expect(csv).toMatchSnapshot();
});

it('baixeResultados deve baixar dois arquivos', () => {
  global.URL.createObjectURL = jest.fn();
  global.URL.revokeObjectURL = jest.fn();
  const sorteados = [
    { 'Inscricao': '123', 'CNPJ': '48797752000151', 'RazaoSocial': 'Produtora xyz' }
  ];
  const log = 'log 123';
  const data = '11/11/2018';
  baixeResultados(sorteados, log, data);
  expect(global.URL.createObjectURL).toHaveBeenCalledTimes(2);
  expect(global.URL.revokeObjectURL).toHaveBeenCalledTimes(2);
});
