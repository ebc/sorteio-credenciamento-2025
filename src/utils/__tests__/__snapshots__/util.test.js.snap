// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gerar string CSV a partir da lista de sorteados 1`] = `
"Lote,Inscrição,CNPJ,Razão Social
1,123,48797752000151,Produtora xyz
"
`;
